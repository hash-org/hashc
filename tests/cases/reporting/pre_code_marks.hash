// stage=parse, run=fail

// Here we are testing that the reporting sub-system
// is correctly adjusting the span labels and widths for 
// any encountered combining characters.

normal := () => {
    // ~ERROR: expected expression
    "aaa" + 
}

crazy := () => {
    // ~ERROR: expected expression
    "äää" + 
}

start := "aaa"; normal_block := ((foo, 
bar)) => {


}

start := "äää"; crazy_block_1 := ((foo, 
bar)) => {

}

start := "äää"; crazy_block_2 := ((foo, 
bar, 
baz, "äää")) => {

}
