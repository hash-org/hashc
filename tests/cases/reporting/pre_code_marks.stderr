error: expected an expression
  --> $DIR/pre_code_marks.hash:9:12
 8 |       // ~ERROR: expected expression
 9 |       "aaa" + 
   |              ^ 
10 |   }

error: expected an expression
  --> $DIR/pre_code_marks.hash:14:15
13 |       // ~ERROR: expected expression
14 |       "äää" + 
   |              ^ 
15 |   }

error: expected a name here
  --> $DIR/pre_code_marks.hash:17:34
15 |    }
16 |    
17 |    start := "aaa"; normal_block := ((foo, 
   |  ___________________________________-
18 | |  bar)) => {
   | |_____- 
19 |    
20 |    
   = help: expected a `identifier`

error: expected a name here
  --> $DIR/pre_code_marks.hash:23:38
21 |    }
22 |    
23 |    start := "äää"; crazy_block_1 := ((foo, 
   |  ____________________________________-
24 | |  bar)) => {
   | |_____- 
25 |    
26 |    }
   = help: expected a `identifier`

error: expected a name here
  --> $DIR/pre_code_marks.hash:28:38
26 |    }
27 |    
28 |    start := "äää"; crazy_block_2 := ((foo, 
   |  ____________________________________-
29 | |  bar, 
30 | |  baz, "äää")) => {
   | |____________- 
31 |    
32 |    }
   = help: expected a `identifier`
